server/container/util/src/test/java/org/apache/james/util/CloseableIteratorTest.java

```diff
@@ -0,0 +1,191 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.util;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.Test;
+
+import com.google.common.collect.ImmutableList;
+
+class CloseableIteratorTest {
+    private static final Closeable THROW_RUNTIME = () -> {
+        throw new RuntimeException();
+    };
+    private static final Closeable THROW_IO_EXCEPTION = () -> {
+        throw new IOException();
+    };
+
+    @Test
+    void closeableIteratorShouldPreserveWrappedIteratorContent() {
+        AtomicBoolean open = new AtomicBoolean(true);
```

Comment:
Why do you need this atomix boolean if you don't test do any assert on its value ?

Reply:
Fast written tests I guess...